gnome-initial-setup/pages/language/eos-test-mode

```diff
@@ -29,25 +29,12 @@ systemctl -q is-active "$extra_unit" && extra_active=true || extra_active=false
 $extra_active && systemctl stop "$extra_unit"
 
 # Mount overlays over any directory that might be written to
-overlay_dirs="bin boot endless etc home lib opt ostree root sbin srv sysroot var"
-for dir in $overlay_dirs; do
-    [ -d /$dir ] || continue
-    # If the directory is a symlink, assume it's pointing to a location
-    # covered by another top level overlay
-    [ -L /$dir ] && continue
-    mkdir -p /run/eos-test/$dir /run/eos-test/$dir-workdir
-    mount -t overlay -o \
-        rw,upperdir=/run/eos-test/$dir,lowerdir=/$dir,workdir=/run/eos-test/$dir-workdir \
-        eos-test-$dir /$dir
-done
+eos-live-boot-overlayfs-setup etc
 
 # Remount the SD card and mount a scratch overlay over it
 if $extra_active; then
     systemctl start "$extra_unit"
-    mkdir -p /run/eos-test/endless-extra /run/eos-test/endless-extra-workdir
-    mount -t overlay -o \
-        rw,upperdir=/run/eos-test/endless-extra,lowerdir=/var/endless-extra,workdir=/run/eos-test/endless-extra-workdir \
-        eos-test-endless-extra /var/endless-extra
+    eos-live-boot-overlayfs-setup var/endless-extra
```

Comment:
I guess this is relying on the `[ -d /run/eos-live/$dir ] && continue` guards in `eos-live-boot-overlayfs-setup` to make all the other mounts idempotent. It might be neater to handle `endless-extra` directly in `eos-live-boot-overlayfs-setup` though of course actual live boots do not have split disk!

Reply:
Maybe we can do something like this - would also help to not call the flatpak overlay setup twice:

```diff
diff --git a/gnome-initial-setup/pages/language/eos-test-mode b/gnome-initial-setup/pages/language/eos-test-mode
index dc1fccae..e11d3218 100644
--- a/gnome-initial-setup/pages/language/eos-test-mode
+++ b/gnome-initial-setup/pages/language/eos-test-mode
@@ -29,12 +29,13 @@ systemctl -q is-active "$extra_unit" && extra_active=true || extra_active=false
 $extra_active && systemctl stop "$extra_unit"
 
 # Mount overlays over any directory that might be written to
-eos-live-boot-overlayfs-setup etc
-
-# Remount the SD card and mount a scratch overlay over it
-if $extra_active; then
+# if there is a sd card remount it and mount a scratch overlay over it
+if $extra_active;
+then
     systemctl start "$extra_unit"
-    eos-live-boot-overlayfs-setup var/endless-extra
+    eos-live-boot-overlayfs-setup etc var/endless-extra
+else
+    eos-live-boot-overlayfs-setup etc
 fi
```

Reply:
I think the point of the current logic (stop extra_unit; overlay var; start extra_unit; overlay var/endless-extra);  is to remount the SD card over the overlayed `/var`, but I don't know if this is actually important.

Reply:
Ok, so we have to special case it then in the eos-live-boot-overlayfs-setup script.

Reply:
I think what we have here may actually work – `eos-live-boot-overlayfs-setup` does check whether `/var/eos-live/$foo` exists and skips overlaying `$foo` if so

Reply:
We decide to leave this as-is for now in the standup -- it should work fine.